src/main.rs

```diff
@@ -30,15 +33,21 @@ impl Service for HelloWorld {
     type Response = Response;
     type Error = hyper::Error;
 
-    type Future = Box<Future<Item=Self::Response, Error=Self::Error>>;
+    type Future = Box<Future<Item = Self::Response, Error = Self::Error>>;
 
     fn call(&self, req: Self::Request) -> Self::Future {
-        // TODO: parse body.
-        let path = String::from(req.uri().path());
         let tx = self.sender.clone();
-        Box::new(tx.send(path)
+        let work = req.body().concat2()
+            // Body to string
+            .map_err(Error::from)
+            .and_then(|v| {
+                // TODO: Parse JSON.
+                str::from_utf8(v.as_ref()).map(String::from).map_err(Error::from)
+            })
+            // Process body
+            .and_then(|body: String| { tx.send(body).map_err(Error::from) })
             .map_err(|error| {
```

Comment:
you still need this?

Reply:
Yes. To go to `hyper::Error`. I'll address this later.